lib/Doctrine/ODM/PHPCR/UnitOfWork.php

```diff
@@ -2860,7 +2868,7 @@ public function findVersionByName($className, $id, $versionName)
 
         $hints = array('versionName' => $versionName, 'ignoreHardReferenceNotFound' => true);
         $frozenDocument = $this->getOrCreateDocument($className, $node, $hints);
-        $this->dm->detach($frozenDocument);
+        //$this->dm->detach($frozenDocument);
```

Comment:
(it no longer needs to be detached - rather these documents are given the STATE_FROZEN state so they are not included for updates etc)